src/VolumeLogger.js

```diff
@@ -59,6 +59,295 @@ module.exports = class VolumeLogger {
+                        minReport.broker.bytesToNetworkPerSec = secReport.broker.bytesToNetworkPerSec
+
+                        minReport.network.avgLatencyMs = secReport.network.avgLatencyMs
+                        minReport.network.bytesToPeersPerSec = secReport.network.bytesToPeersPerSec
+                        minReport.network.bytesFromPeersPerSec = secReport.network.bytesFromPeersPerSec
+                        minReport.network.connections = secReport.network.connections
+
+                        if (metricsReport.metrics['broker/cassandra']) {
+                            minReport.storage.bytesWrittenPerSec = secReport.storage.bytesWrittenPerSec
+                            minReport.storage.bytesReadPerSec = secReport.storage.bytesReadPerSec
+                        }
+                    } else {
+                        minReport.broker.messagesToNetworkPerSec = throtheledAvg(minReport.broker.messagesToNetworkPerSec, secReport.broker.messagesToNetworkPerSec)
+                        minReport.broker.bytesToNetworkPerSec = throtheledAvg(minReport.broker.bytesToNetworkPerSec, secReport.broker.bytesToNetworkPerSec)
+
+                        minReport.network.avgLatencyMs = throtheledAvg(minReport.network.avgLatencyMs, secReport.network.avgLatencyMs)
+                        minReport.network.bytesToPeersPerSec = throtheledAvg(minReport.network.bytesToPeersPerSec, secReport.network.bytesToPeersPerSec)
+                        minReport.network.bytesFromPeersPerSec = throtheledAvg(minReport.network.bytesFromPeersPerSec, secReport.network.bytesFromPeersPerSec)
+                        minReport.network.connections = throtheledAvg(minReport.network.connections, secReport.network.connections)
+
+                        if (metricsReport.metrics['broker/cassandra']) {
+                            minReport.storage.bytesWrittenPerSec = throtheledAvg(minReport.storage.bytesWrittenPerSec, secReport.storage.bytesWrittenPerSec)
+                            minReport.storage.bytesReadPerSec = throtheledAvg(minReport.storage.bytesReadPerSec, secReport.storage.bytesReadPerSec)
+                        }
+                    }
+
+                    securePublish(
+                        this.streamIds.secStreamId,
+                        secReport
+                    )
+
+                    const now = Date.now()
+
+                    if (sec === 60 || (minReport.timestamp + (60 * 1000) - now) < 0) {
+                        minReport.timestamp = secReport.currentTime
+                        sec = 0
+
+                        securePublish(
+                            this.streamIds.minStreamId,
+                            minReport
+                        )
+
+                        // let's check, once a minute, if it's time to run hour and/or daily reports
+
+                        // get the last msg to check if it's been an hour
+
+                        const lastHourReports = await getResend(this.streamIds.hourStreamId, 1)
```

Comment:
I'm wondering if it would be enough to check this in the beginning of the program, and then set this to happen hourly thereafter?

Reply:
Hmm.. not necessarily though if we are unable to push an hourly report. Needs a discussion I think.